noxfile.py

```diff
@@ -9,17 +10,34 @@
 # default actions to be run if nothing is explicitly specified with the -s option
 nox.options.sessions = ["fix"]
 
+
 @nox.session(name="generate-api", python=False)
 def generate_api(session: Session):
     """
     Call openapi-python-client to generate the client api based on the
     openapi specification for Exasol SaaS in JSON format, see
     https://github.com/openapi-generators/openapi-python-client.
+
+    By default run generator silently, Except for CI build, which is detected
+    by environment variable ``CI``, see
+    https://docs.github.com/en/actions/learn-github-actions/variables.
+    #default-environment-variables.
     """
+    silent = False if os.environ.get("CI") else True
```

Comment:
```suggestion
    silent = "CI" in os.environ
```

this will actually fail, when it is not set

Reply:
Actually, the `.get()` did not fail, when the variable was unset, see https://python-reference.readthedocs.io/en/latest/docs/dict/get.html. 

But I prefer your proposal as it is shorter.